### PR TITLE
[SPARK-42612][CONNECT][PYTHON][TESTS] Enable more parity tests related to functions

### PR DESCRIPTION
**Pull request by ueshin**
### What changes were proposed in this pull request?

Enables more parity tests related to `functions`.

### Why are the changes needed?

There are still some more tests we should enable.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Modified/enabled related tests.

### CODE REVIEW

File: python/pyspark/sql/tests/test_functions.py [hunk 1]
```diff
@@ -318,19 +318,29 @@ def test_math_functions(self):
...
+                    [n * -0.2 for n in range(10)],
+                    df.select(inv(trig(df.b))).collect(),
+                )
 
-        cols = ["a", functions.col("a")]
+        from pyspark.sql import functions
 
-        for f, alias in funs:
-            for c in cols:
-                self.assertIn(f"{alias}(a)", repr(f(c)))
```
Comment (ueshin) [inline python/pyspark/sql/tests/test_functions.py:333]: The upper case name needs to communicate with server side. Do we still want to match this?

**Review by HyukjinKwon — APPROVED** [review]

File: python/pyspark/sql/tests/test_functions.py [hunk 2]
```diff
@@ -578,9 +588,13 @@ def test_approxQuantile(self):
         self.assertRaises(TypeError, lambda: df.stat.approxQuantile(["a", 123], [0.1, 0.9], 0.1))
 
     def test_sorting_functions_with_column(self):
-        from pyspark.sql import functions
         from pyspark.sql.column import Column
 
+        self.check_sorting_functions_with_column(Column)
+
+    def check_sorting_functions_with_column(self, tpe):
+        from pyspark.sql import functions
```
Comment (HyukjinKwon) [inline python/pyspark/sql/tests/test_functions.py:596]: no biggie but I believe we can just import this on the top

Comment (ueshin) [reply python/pyspark/sql/tests/test_functions.py:596]: Yeah, but it's imported in each test for some reason. Shall we do it in a separate PR?

Comment (HyukjinKwon) [reply python/pyspark/sql/tests/test_functions.py:596]: 🆗

Comment (ueshin) [reply python/pyspark/sql/tests/test_functions.py:596]: oh, actually, it's already imported at the top of this file.

**Review by zhengruifeng — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.4.

**Pull request closed without merging**